sched/pthread/pthread_condwait.c

```diff
@@ -94,6 +94,7 @@ int pthread_cond_wait(FAR pthread_cond_t *cond, FAR pthread_mutex_t *mutex)
       sinfo("Give up mutex / take cond\n");
 
       flags = enter_critical_section();
```

Comment:
Do we still need critical section here?

Reply:
Yes, it needs since the critical section ls leaved at the end of sched_lock.

Reply:
Initial sched lock/unlock was replaced by a critical section in https://github.com/apache/nuttx/pull/9067, so I'm wondering maybe now we just need to revert that commit?

Reply:
We need both here:

1. the critical section is required for SMP case
2. the sched lock is required to avoid the current thread switch out in the middle of operation